### PR TITLE
chore: Split sink into multiple files

### PR DESCRIPTION
**Pull request by blt**
This commit adjusts the structure of the datadog logs sink module into multiple
files. I have some fairly large changes I'd like to experiment with here and
it's convenient to have things split up. Each file is more or less
single-concept. I have maintained the external API of this module.

Signed-off-by: Brian L. Troutwine <brian@troutwine.us>

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]
I'm guessing the answer is no - but would there be any benefit to moving these to be tower based?

**Comment by spencergilbert** [comment]
Rerunning jobs as it failed during linking 💤

**Comment by blt** [comment]
> I'm guessing the answer is no - but would there be any benefit to moving these to be tower based?

That is, in point of fact, my current hunch. I think if nothing else we'll have an easier time building sinks in a composable way.

File: src/sinks/datadog/logs/tests.rs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+use indoc::indoc;
+use pretty_assertions::assert_eq;
+use std::sync::Arc;
+use vector_core::event::Event;
+use vector_core::event::{BatchNotifier, BatchStatus};
+
+#[test]
+fn generate_config() {
+    crate::test_util::test_generate_config::<DatadogLogsConfig>();
+}
```
Comment (jszwedko) [inline src/sinks/datadog/logs/tests.rs:23]: I could see this test living in `config.rs` and `test.rs` just having the integration tests.

**Review by jszwedko — APPROVED** [review]

File: src/sinks/datadog/logs/tests.rs (hunk 1 above)
Comment (blt) [reply src/sinks/datadog/logs/tests.rs:23]: Ah yeah, I did just kind of move all the tests together in one block. I'll scoot them around to their appropriate homes.

**Pull request merged**